phac-nml/irida-next
# Transfer samples service

## What does this PR do and why?
The sample transfer service moves samples between projects. This functionality is restricted to owners.
Pulled from https://github.com/phac-nml/irida-next/pull/90.

## Screenshots or screen recordings
N/A

## How to set up and validate locally
1.  Call the sample transfer service within the rails console:
`Samples::TransferService.new(project, user).execute(sampleTransfer)`
For examples, see test/services/samples/transfer_service_test.rb.
3. Verify only owners can call the service.
4. Verify you can't move samples to the same project.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
